project2/screens/search.js

```diff
@@ -1,131 +1,72 @@
-        if (this.state.searchString !== prevState.searchString) {
-            this.getMovies(this.state.searchString)
+import SearchListItem from "./Search/ListItem";
+import s from "../Styles";
+
+const Search = () => {
+  const [searchString, setSearchString] = React.useState("");
+  const [movies, setMovies] = React.useState([]);
+  const [totalResults, setTotalResults] = React.useState(0);
+  const [totalPages, setTotalPages] = React.useState(0);
+  const [nextPage, setNextPage] = React.useState(0);
+
+  React.useEffect(() => {
+    const getMovies = async () => {
+      const { result, totalResults, totalPages } = await searchMovies(
+        searchString
+      );
+
+      if (result) {
+        setMovies(result);
+        setTotalResults(totalResults);
+        setTotalPages(totalPages);
+
+        if (totalPages > 1) {
+          setNextPage(2);
         }
-    }
-
-    updateSearch = searchString => {
-        this.setState({
-            searchString: searchString,
-            movies: "",
-            totalResults: 0,
-            totalPages: 0,
-            nextPage: 0
-        })
-    }
-
-    movieSelected = (id) => {
-        this.props.navigation.navigate("Movie", { "MovieID": id })
-    }
-
-    displayMore = async () => {
-        if (this.state.totalPages === 1) return
-        if (this.state.nextPage === this.state.totalPages) return
-
-        const results = await getMoviePage(this.state.searchString, this.state.nextPage)
-
-        if (results !== undefined) {
-            this.setState(prevState => ({ movies: [...prevState.movies, ...results] }));
+      }
+    };
 
-            if (this.state.nextPage < this.state.totalPages) {
-                this.setState({ nextPage: this.state.nextPage++ })
-            }
-        }
+    if (searchString !== "") {
+      getMovies();
     }
+  }, [searchString]);
 
-    render() {
-        return (
-            <View style={styles.appContainer}>
-                <Text>Find a movie</Text>
-                <TextInput
-                    style={styles.searchInput}
-                    value={this.state.searchString}
-                    onChangeText={this.updateSearch}
-                />
-
-                <Text>Results found: {this.state.totalResults}</Text>
-                <FlatList
-                     onEndReached={this.displayMore}
-                    data={this.state.movies}
-                    renderItem={renderItem(this.movieSelected)}
-                    keyExtractor={(item) => item.id}
-               
-                />
+  const displayMore = async () => {
+    if (totalPages > 1 || nextPage < totalPages) {
+      const results = await getMoviePage(searchString, nextPage);
 
-            </View>
-        )
-    }
-}
-
-const styles = StyleSheet.create({
-    appContainer: {
-        flex: 1,
-        backgroundColor: "#AAAAAF",
-        paddingTop: Constants.statusBarHeight,
-        width: "100%",
+      if (Array.isArray(results)) {
```

Comment:
`results` should always be an array if we've fetched more movies, so make the check be this to allow concatenating more movies.